### PR TITLE
Label project with claim - OSD-6972

### PR DESCRIPTION
**Pull request by sulmer-redhat**
### What type of PR is this? 
_(bug/feature/cleanup/docs/design/test/chore/refactor..)_

### Which issue(s) this PR fixes(can be a reference to existing issue or jira/bz):

_Fixes #_

### Pre-checks:
- [ ] manually tested latest changes against crc/k8s
- [ ] run the `make coverage` command to generate new calculated coverage

### CODE REVIEW

**Review by georgettica — COMMENTED** [review]
had a good talk and these are the only nits I found

**Comment by georgettica** [comment]
fixed the mockgen issue with https://github.com/golang/mock/issues/415

File: pkg/gcpclient/client.go [hunk 1]
```diff
@@ -154,11 +152,39 @@ func (c *gcpClient) GetProject(projectID string) (*cloudresourcemanager.Project,
...
+//	labelsMap["claim_name"] = claimName
+	if project.Labels == nil {
+		project.Labels = make(map[string]string)
+	}
+	project.Labels["claim_name"] = claimName
+
+	// operation, err := c.cloudResourceManagerClient.Projects.Update(projectID, &project).Do()
+	operation, err := c.cloudResourceManagerClient.Projects.Update(project.ProjectId, project).Do()
+	if err != nil {
+		return &cloudresourcemanager.Operation{}, fmt.Errorf("gcpclient.CreateProject.Projects.Update %v", err)
```
[the remark this answers is not in the document]
Comment (sulmer-redhat) [reply pkg/gcpclient/client.go]: No reason, you're right :) changed.

File: pkg/gcpclient/client.go [hunk 2]
```diff
@@ -154,11 +152,39 @@ func (c *gcpClient) GetProject(projectID string) (*cloudresourcemanager.Project,
...
+// CreateProjectLabels creates the claimName label on a project
+func (c *gcpClient) CreateProjectLabels(project *cloudresourcemanager.Project, claimName string) (*cloudresourcemanager.Operation, error){
+	log.V(2).Info("Started gcpClient.CreateProjectLabels")
+	
+//	labelsMap := make(map[string]string)
+//	labelsMap["claim_name"] = claimName
+	if project.Labels == nil {
+		project.Labels = make(map[string]string)
+	}
+	project.Labels["claim_name"] = claimName
```
[the remark this answers is not in the document]
Comment (sulmer-redhat) [reply pkg/gcpclient/client.go]: Yes, we will force overwrite the label.

**Comment by NautiluX** [comment]
I think you're using a different version of mockgen then expected by the CI. Can you run `./boilerplate/_src/container-make GO111MODULE=on go-generate`?

File: pkg/controller/projectreference/projectreference_adapter.go [hunk 3]
```diff
@@ -560,6 +570,25 @@ func (r *ReferenceAdapter) configureAPIS() error {
 	return nil
 }
 
+func (r *ReferenceAdapter) ConfigureProjectLabel() (gcputil.OperationResult, error) {
```
Comment (NautiluX) [inline pkg/controller/projectreference/projectreference_adapter.go:573]: we should not do this for CCS projects. I think when creating projects we don't set it, but this part may need a ccs check.

File: pkg/gcpclient/client.go [hunk 4]
```diff
@@ -154,12 +152,35 @@ func (c *gcpClient) GetProject(projectID string) (*cloudresourcemanager.Project,
...
+	project.Labels = labels
+
+	// operation, err := c.cloudResourceManagerClient.Projects.Update(projectID, &project).Do()
+	operation, err := c.cloudResourceManagerClient.Projects.Update(project.ProjectId, project).Do()
+	if err != nil {
+		return fmt.Errorf("gcpclient.CreateProject.Projects.Update %v", err)
+	}
+	time.Sleep(3 * time.Second) //Wait 3 seconds to make it more probable the project is updated after returning
+	fmt.Println(operation)
+	// "operation" here is of type Project and not Operation
```
Comment (NautiluX) [inline pkg/gcpclient/client.go:168]: why is it called operation then?

File: pkg/gcpclient/client.go [hunk 5]
```diff
@@ -154,12 +152,35 @@ func (c *gcpClient) GetProject(projectID string) (*cloudresourcemanager.Project,
 	return project, nil
 }
 
+// CreateProjectLabels creates the claimName label on a project
+func (c *gcpClient) CreateProjectLabels(project *cloudresourcemanager.Project, labels map[string]string) error {
+	log.V(2).Info("Started gcpClient.CreateProjectLabels")
+
+	project.Labels = labels
+
+	// operation, err := c.cloudResourceManagerClient.Projects.Update(projectID, &project).Do()
```
Comment (NautiluX) [inline pkg/gcpclient/client.go:161]:
```suggestion
```
Remove if not needed

File: pkg/gcpclient/client.go [hunk 6]
```diff
@@ -154,12 +152,35 @@ func (c *gcpClient) GetProject(projectID string) (*cloudresourcemanager.Project,
...
+	operation, err := c.cloudResourceManagerClient.Projects.Update(project.ProjectId, project).Do()
+	if err != nil {
+		return fmt.Errorf("gcpclient.CreateProject.Projects.Update %v", err)
+	}
+	time.Sleep(3 * time.Second) //Wait 3 seconds to make it more probable the project is updated after returning
+	fmt.Println(operation)
+	// "operation" here is of type Project and not Operation
+	// return operation, nil
```
Comment (NautiluX) [inline pkg/gcpclient/client.go:169]:
```suggestion
```
remove if not needed

**Review by NautiluX — CHANGES_REQUESTED** [review]

File: pkg/controller/projectreference/projectreference_adapter.go (hunk 3 above)
Comment (sulmer-redhat) [reply pkg/controller/projectreference/projectreference_adapter.go:573]: Added isCCS check and fixed tests.

File: pkg/gcpclient/client.go (hunk 4 above)
Comment (sulmer-redhat) [reply pkg/gcpclient/client.go]: Deleted

**Review by NautiluX — COMMENTED** [review]
/lgtm
/hold
unhold when you're ready to merge

**Review by NautiluX — APPROVED** [review]

**Pull request merged**